autoPyTorch/pipeline/components/preprocessing/tabular_preprocessing/feature_preprocessing/LibLinearSVCPreprocessor.py

```diff
@@ -0,0 +1,141 @@
+from typing import Any, Dict, Optional
+
+from ConfigSpace.configuration_space import ConfigurationSpace
+from ConfigSpace.forbidden import (
+    ForbiddenAndConjunction,
+    ForbiddenEqualsClause,
+)
+from ConfigSpace.hyperparameters import (
+    CategoricalHyperparameter,
+    UniformFloatHyperparameter,
+    UniformIntegerHyperparameter
+)
+
+import numpy as np
+
+from sklearn.base import BaseEstimator
+from sklearn.feature_selection import SelectFromModel
+from sklearn.svm import LinearSVC
+
+
+from autoPyTorch.datasets.base_dataset import BaseDatasetPropertiesType
+from autoPyTorch.pipeline.components.preprocessing.tabular_preprocessing.feature_preprocessing. \
+    base_feature_preprocessor import autoPyTorchFeaturePreprocessingComponent
+from autoPyTorch.utils.common import HyperparameterSearchSpace, add_hyperparameter, get_hyperparameter
+
+
+class LibLinearSVCPreprocessor(autoPyTorchFeaturePreprocessingComponent):
+    """
+    Selects features based on importance weights using svm classifier
+    """
+    def __init__(self, dual: bool = False, penalty: str = "l1",
+                 loss: str = "squared_hinge", tol: float = 1e-4,
+                 C: float = 1, multi_class: str = "ovr",
+                 intercept_scaling: int = 1, fit_intercept: bool = True,
+                 random_state: Optional[np.random.RandomState] = None):
+
+        self.dual = dual
+        self.penalty = penalty
+        self.loss = loss
+        self.multi_class = multi_class
+        self.intercept_scaling = intercept_scaling
+        self.fit_intercept = fit_intercept
+        self.tol = tol
+        self.C = C
+
+        super().__init__(random_state=random_state)
+
+    def get_components_kwargs(self) -> Dict[str, Any]:
+        """
+        returns keyword arguments required by the feature preprocessor
+
+        Returns:
+            Dict[str, Any]: kwargs
+        """
```

Comment:
I think `@property` is better, because we can use it as if it is an attribuite?

Reply:
It is actually a matter of taste and I just think that property is better.
This is something which you prefer:
```python
from XXX import func

class C:
    def __init__(self):
        self.a = 1
        self.b = 2.5
        self.c = "test"
        self.d = True

    @property
    def attr_dict(self):
        return dict(
            a=self.a,
            b=self.b,
            c=self.c,
            d=self.d
        )

    def get_attr_dict(self):
        return dict(
            a=self.a,
            b=self.b,
            c=self.c,
            d=self.d
        )

    def test_func(self):
        # Given the methods, you prefer which?
        func(**self.get_attr_dict())  # <== I just feel strange because we have ** in front of the call of a method
        func(**self.attr_dict)  # <== We can handle as if it is a dict
```

I also feel strange when I see `get_dict()["key"]` by the similar reason

Reply:
I think the function already has kwargs in its name and that makes what we want to do clear enough. I am opposed to having it as a `property` because this is for our internal use only, I'd rather add an `_` in front of the name of the functions which I will do before we resolve this comment.